demo/admin/src/products/categories/ProductCategoriesPage.tsx

```diff
@@ -45,7 +56,69 @@ const ProductCategoriesPage: React.FC = () => {
+                                        })}
+                                    >
+                                        <Box sx={{ height: "100vh" }}>
+                                            <ProductsGrid
+                                                // TODO refresh ProductsGrid after saving assign-products
+                                                toolbarAction={
+                                                    <Button
+                                                        startIcon={<AddIcon />}
+                                                        onClick={() => setIsOpen(true)}
+                                                        variant="contained"
+                                                        color="primary"
+                                                    >
+                                                        <FormattedMessage
+                                                            id="products.editProductCategory.assignProducts"
+                                                            defaultMessage="Assign Products"
+                                                        />
+                                                    </Button>
+                                                }
+                                                filter={{ category: { equal: selectedId } }}
+                                            />
+                                            <SaveBoundary onAfterSave={() => setIsOpen(false)}>
+                                                <Dialog open={isOpen} onClose={handleCloseDialog} fullWidth maxWidth="xl">
+                                                    <DialogTitle>
+                                                        <FormattedMessage
+                                                            id="products.editProductCategory.assignProducts"
+                                                            defaultMessage="Assign Products"
+                                                        />
+                                                    </DialogTitle>
+                                                    <DialogContent
+                                                        sx={{
+                                                            height: "70vh",
+                                                            padding: 0,
+                                                            paddingTop: "0 !important" /* is connected to title-style */,
+                                                        }}
+                                                    >
+                                                        <AssignProductsGrid productCategoryId={selectedId} />
+                                                    </DialogContent>
+                                                    <DialogActions>
+                                                        {/* TODO Missing close-dialog-unsaved-changes-check */}
```

Comment:
EditDialog had some issues in this situation. @thomasdax98 told me to use something else because EditDialog was not meant to be used for something like that.